lib/uart.cpp

```diff
@@ -187,10 +181,34 @@ void Uart::write(Uart::Address register_address, uint8_t data) {
     }
 }
 
-bool Uart::interrupt_pending() const {
+void Uart::update_receive() {
+    uint8_t data;
+    ssize_t bytesRead = ::read(this->receive_file_fd, &data, 1);
+    if(bytesRead > 0){
+            receive_queue.push(data);
+            this->m_isr |= 0b1;
+    }else if(receive_queue.size() == 0){
+        this->m_isr &= 0b11111110;
```

Comment:
Kind of a stylistic thing, honestly it doesn't both me that much as is.

Reply:
Yeah obviously it doesn't matter that much. Reason I make the suggestion is because:
- In my experience, the convention for bit clearing is always `&= ~`.
- The way of doing it right now won't work if the width of `m_isr` changes. Not that it ever would but I'd always rule in favor of good practice

Reply:
Oh cool I didn't think about the size changing in the future. That's why people do that, I always wondered!

Reply:
Thank you. I've addressed this in my most recent commit.